client/lib/features/admin/presentation/views/data_tab.dart

```diff
@@ -0,0 +1,529 @@
+                borderRadius: BorderRadius.circular(16),
+                child: Padding(
+                  padding: const EdgeInsets.all(16.0),
+                  child: SingleChildScrollView(
+                    child: Column(
+                      children: [
+                        FutureBuilder<List<Widget>>(
+                          future: Future.wait([
+                            // Build the event rows
+                            for (int i = _currentStartIndex;
+                                i < _currentStartIndex + 5 && i < events.length;
+                                i++)
+                              _buildEventRow(
+                                index: i,
+                                event: events[i],
+                                isMobile: isMobile,
+                              ),
+                          ]),
+                          builder: (context, snapshot) {
+                            if (snapshot.hasData) {
+                              return Column(children: snapshot.data!);
+                            }
+                            if (snapshot.hasError) {
+                              return Text(
+                                context.l10n.errorOccurred,
+                                style: context.theme.textTheme.bodyLarge,
+                              );
+                            }
+                            return Center(
+                              child: CircularProgressIndicator(),
+                            );
+                          },
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.all(16.0),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Text(
+                    '${events.length} ${context.l10n.events}',
+                    style: context.theme.textTheme.bodyMedium!.copyWith(
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                  Row(
+                    children: [
+                      IconButton(
+                        onPressed: _currentStartIndex == 0
+                            ? null
+                            : () {
+                                setState(() {
+                                  _currentStartIndex -= 5;
+                                });
+                              },
+                        icon: Icon(Icons.arrow_back_rounded),
+                      ),
+                      Text(
+                        '${_currentStartIndex + 1} - ${events.length > 5 ? _currentStartIndex + 5 : events.length} of ${events.length}',
+                        style: context.theme.textTheme.bodyMedium!.copyWith(
+                          fontWeight: FontWeight.bold,
+                        ),
```

Comment:
```suggestion
                        '${_currentStartIndex + 1} - ${events.length > 5 ? min(_currentStartIndex + 5, events.length) : events.length} of ${events.length}',
```

Fixes issue with the index exceeding the total num of events!